plugins/modem-manager/fu-firehose-updater.c

```diff
@@ -76,6 +77,32 @@ fu_firehose_updater_log_message(const gchar *action, GBytes *msg)
 	g_debug("%s: %.*s", action, (gint)msg_size, msg_strsafe);
 }
 
+static GBytes *
+fu_firehose_read(FuFirehoseUpdater *self, guint timeout_ms, FuIOChannelFlags flags, GError **error)
+{
+	if (self->sahara != NULL) {
+		GByteArray *bytearr = NULL;
+
+		bytearr = fu_sahara_loader_qdl_read(self->sahara, error);
+		return g_byte_array_free_to_bytes(bytearr);
```

Comment:
i think you need to check if bytearr is NULL and return NULL before the call to g_byte_array_free_to_bytes

Reply:
added `bytearr == NULL` check